layer/layer_store.go

```diff
@@ -316,8 +316,7 @@ func (ls *layerStore) registerWithDescriptor(ts io.Reader, parent ChainID, descr
 	defer ls.layerL.Unlock()
 
 	if existingLayer := ls.getWithoutLock(layer.chainID); existingLayer != nil {
-		// Set error for cleanup, but do not return the error
-		err = errors.New("layer already exists")
+		logrus.Warnf("Layer %q already exists", layer.chainID)
```

Comment:
Pretty sure this will break something. If you look at the comment, the error is set for the `cleanup`, meaning it's getting picked-up by the `defer` func. We should update the comment so the intent is more clear.
/cc @dmcgowan